src/hmem.c

```diff
@@ -170,3 +175,35 @@ void ofi_hmem_cleanup(void)
 			hmem_ops[iface].cleanup();
 	}
 }
+
+enum fi_hmem_iface ofi_get_hmem_iface(const void *addr)
+{
+	enum fi_hmem_iface iface;
+	int ret;
+
+	/* Since the iface of the given address is assumed to be unknown by the
+	 * caller, HMEM ifaces may not have been initialized. Perform one time
+	 * initialization of all HMEM ifaces. Only successfully initialized
+	 * HMEM ifaces will be queried to see if they own a given address.
+	 */
+	if (!global_hmem_init) {
+		for (iface = 0; iface < ARRAY_SIZE(hmem_ops); iface++) {
+			ret = ofi_hmem_init(iface);
+			if (ret != FI_SUCCESS)
+				FI_DBG(&core_prov, FI_LOG_CORE,
+				       "Failed to initialize HMEM iface %d\n",
+				       iface);
+		}
+
+		global_hmem_init = true;
+	}
```

Comment:
There's got to be a better place to initialize everything than during the first ofi_get_hmem_iface call that a provider makes.

Reply:
Would adding a global init to fi_ini() be better? Or we could make a requirement to have providers initialize all hmem interfaces prior to performing data transfer operations?

Reply:
I would update fi_ini().  You may also need to update providers for DL builds to call the same function.

Reply:
I'll open a new PR for HMEM initialization in fi_ini() and DL builds.